packages/loader/container-loader/src/loader.ts

```diff
@@ -260,7 +260,9 @@ export interface ILoaderServices {
  * Subset of IDocumentStorageService which only supports createBlob() and readBlob(). This is used to support
  * blobs in detached containers.
  */
- export type IDetachedBlobStorage = Pick<IDocumentStorageService, "createBlob" | "readBlob">;
+export type IDetachedBlobStorage = Pick<IDocumentStorageService, "createBlob" | "readBlob"> & {
+    size: number;
```

Comment:
I think we want to improve here. User might have deleted some content in between local serializations resulting in no blobs used any more. I.e. blobs might have been saved earlier, but they are no longer used, or alternative content was saved. I think we need somehow to consult runtime here.

I do not have good suggestion on how to implement it, as pull model (loader pulls data from runtime) does not work well here. If we had push model (runtime was asking loader to create empty file, upload blobs, upload summary), it would work a bit better, I think.

Reply:
>User might have deleted some content in between local serializations resulting in no blobs used any more. I.e. blobs might have been saved earlier, but they are no longer used, or alternative content was saved.

How would we know this? We normally rely on GC for this information, is it different when detached?

Reply:
It depends.

You are right that we may chose this direction of simply not involving runtime and coping over content of IDetachedBlobStorage into real storage.

I was thinking more of how summary blobs work today, where each summary produces new set of blobs, and while there is reuse (can reference blobs from previous summary), the rest is a new set of blobs. In detached state, given that we do not have notion of summary tree reuse, I was thinking an app may chose to push more content (including DDS summary blobs) into IDetachedBlobStorage to increase reuse between local summaries (serialized containers) by reusing same blobs if DDS did not change from serialization to serialization.

I think we should stick with simpler design for now and re-evaluate if new requirements pop up. I.e. just push all the blobs into storage.